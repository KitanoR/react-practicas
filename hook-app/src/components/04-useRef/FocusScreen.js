import React, { useRef } from 'react'
import '../useEffect/effects.css';

export const FocusScreen = () => {
    const inputRef = useRef();
    const handleClick = () => {
        inputRef.current.select();
    }
    return (
        <div>
            <h1>Focus screen</h1>
            <hr />
            <input
                ref={inputRef}
                className="form-control"
                placeholder="Su nombre"
            />
            <button 
                onClick={ handleClick }
                className="btn btn-primary mt-5">
                Focus
            </button>
        </div>
    )
}
