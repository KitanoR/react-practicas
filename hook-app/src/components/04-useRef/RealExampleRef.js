import React, { useState } from 'react'
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';
import '../useEffect/effects.css';

export const RealExampleRef = () => {
    const [show, setShow] = useState(false);
    return (
        <div>
            <h1>Real example ref</h1>
            <hr />

            { show && <MultipleCustomHooks /> }
            <button
                onClick={() => setShow(!show)}
                className="btn btn-primary mt-5"
            >
                Show / hide
            </button>
        </div>
    )
}
