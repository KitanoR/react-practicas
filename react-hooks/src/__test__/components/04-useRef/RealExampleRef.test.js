import React  from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';
import '@testing-library/jest-dom';


describe('Pruebas para el componente <RealExampleRef />', () => {
    const wrapper = shallow(<RealExampleRef />);
    test('Comprobando que se muestra', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });
    test('Debe mostrar el comonente <MultipleCustomHooks />', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
});