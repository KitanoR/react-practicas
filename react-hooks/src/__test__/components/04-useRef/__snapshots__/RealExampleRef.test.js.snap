// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <RealExampleRef /> Comprobando que se muestra 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
